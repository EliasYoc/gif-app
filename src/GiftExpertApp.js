import { useState } from "react";
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GiftExpertApp = ({ categoris = [] }) => {
  const [categories, setCategories] = useState(categoris);
  return (
    <div>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((el) => (
          <GifGrid key={el} category={el} />
        ))}
      </ol>
      <hr />
    </div>
  );
};

export default GiftExpertApp;
