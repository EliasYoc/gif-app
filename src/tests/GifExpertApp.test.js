import { shallow } from "enzyme";
import GiftExpertApp from "../GiftExpertApp";

describe("pruebas <GifExpertApp/>", () => {
  test("debe ejecutar correctamente", () => {
    const wrapper = shallow(<GiftExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar la lista categorias", () => {
    const categories = ["loki", "dragon ball"];
    const wrapper = shallow(<GiftExpertApp categoris={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
