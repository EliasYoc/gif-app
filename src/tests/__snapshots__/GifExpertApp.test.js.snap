// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas <GifExpertApp/> debe ejecutar correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
  <hr />
</div>
`;

exports[`pruebas <GifExpertApp/> debe mostrar la lista categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="loki"
      key="loki"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
  <hr />
</div>
`;
